oauth2-useragent-core/src/main/java/com/microsoft/alm/oauth2/useragent/JavaFxProvider.java

```diff
@@ -27,30 +28,47 @@ protected JavaFxProvider() {
 
     @Override public List<String> checkRequirements() {
 
-        return checkRequirements(JAVA_RUNTIME_VERSION, potentialJavaFxJarLocations, OS_NAME, ENV_DISPLAY);
+        return checkRequirements(JAVA_RUNTIME_VERSION, potentialJavaFxJarLocations, OS_NAME, OS_VERSION, ENV_DISPLAY);
     }
 
-    static ArrayList<String> checkRequirements(final String javaRuntimeVersion, final File[] potentialJavaFxJarLocations, final String osName, final String displayVariable) {
-        final ArrayList<String> requirements = new ArrayList<String>(REQUIREMENTS);
-        final Version version = Version.parseJavaRuntimeVersion(javaRuntimeVersion);
+    static ArrayList<String> checkRequirements(final String javaRuntimeVersionString, final File[] potentialJavaFxJarLocations, final String osName, final String osVersionString, final String displayVariable) {
+        final ArrayList<String> requirements = new ArrayList<String>();
+        final Version javaVersion = Version.parseJavaRuntimeVersion(javaRuntimeVersionString);

```

Comment:
> It looks like this method can throw an exception.

Which one?  `parseJavaRuntimeVersion()`?  The value comes from that of the `java.runtime.version` system property, so if it's been tampered with, throwing an exception seems fine.